src/simulator.ts

```diff
@@ -88,13 +89,51 @@ export class Simulator {
 
   public async cancel() {
     this.cancelToken = true;
+    await this.delay(1000);
+  }
+
+  public async telemetry(eventName: string, status: boolean, properties?: { [key: string]: string; }) {
+    const iotHubConnectionString = status ? await Utility.getConnectionString(
+      Constants.IotHubConnectionStringKey,
+      Constants.IotHubConnectionStringTitle,
+      false,
+    ) : undefined;
+    if (eventName === Constants.SimulatorLaunchEvent) {
+      TelemetryClient.sendEvent(eventName, {
+        status: status ? "Succeeded" : "Failed",
+        reason: status ? undefined : properties.error,
+        quitWhenProcessing: this.isProcessing() ? "true" : "false",
+      }, iotHubConnectionString);
+    } else if (eventName === Constants.SimulatorSendEvent) {
+      if (status) {
+        TelemetryClient.sendEvent(eventName, {
+          status: "Succeeded",
+          devices: "" + properties.deviceConnectionStrings.length,
+          numbers: "" + properties.numbers,
```

Comment:
1. What does the numbers here mean?
2. Do not record message body

Reply:
1. `numbers` means how many iterations for each device, changed to `IterationsPerDevice` to avoid misunderstanding.
2. No message will be recorded now. Only record the body type, Dummy JSON or Plain Text.